src/controllers/regulation.controller.ts

```diff
@@ -0,0 +1,105 @@
+import { Context } from "koa";
+import logger from "../logger";
+import { getErrorStatusCode } from "../v0/util";
+import { client as errNotificationClient } from "../util/errorNotifier";
+
+// TODO: refactor this class to new format
+export default class RegulationController {
+  // TODO: USe stored state
+  private static getDeletionUserHandler(version, dest) {
+    return `../${version}/destinations/${dest}/deleteUsers`;
+  }
+
+  // TODO: Use new format
+  private static getReqMetadata(ctx: Context) {
+    try {
+      const reqBody = ctx.request.body;
+      return { destType: reqBody?.destType };
+    } catch (error) {
+      // Do nothing
+    }
+    return {};
+  }
+
+  // TODO: use new format
+  private static getCommonMetadata(ctx) {
+    // TODO: Parse information such as
+    // cluster, namespace, etc information
+    // from the request
+    return {
+      namespace: "Unknown",
+      cluster: "Unknown"
+    };
+  }
+  public static async handleUserDeletion(ctx: Context) {
+    const getRudderDestInfo = () => {
+      try {
+        const rudderDestInfoHeader = ctx.get("x-rudder-dest-info");
+        const destInfoHeader = JSON.parse(rudderDestInfoHeader);
+        if (!Array.isArray(destInfoHeader)) {
+          return destInfoHeader;
+        }
+      } catch (error) {
+        logger.error(
+          `Error while getting rudderDestInfo header value: ${error}`
+        );
+      }
+      return {};
+    };
+
+    const { body } = ctx.request;
+    const respList = [];
+    const rudderDestInfo = getRudderDestInfo();
+    let response;
+    await Promise.all(
```

Comment:
Avoid such logic as it is hard to guarantee order

Reply:
This is still to be refactored didn't touch much of regulation flow will take it up

Reply:
This is handled now with refactoring work